components/ntp_background_images/browser/ntp_background_images_service.h

```diff
@@ -58,14 +87,53 @@ class NTPBackgroundImagesService {
                            IsActiveOptedIn);
   FRIEND_TEST_ALL_PREFIXES(NTPBackgroundImagesViewCounterTest,
                            ActiveInitiallyOptedIn);
+  FRIEND_TEST_ALL_PREFIXES(NTPBackgroundImagesSourceTest, BasicTest);
 
-  void OnComponentReady(const base::FilePath& installed_dir);
-  void OnGetPhotoJsonData(const std::string& photo_json);
-  void NotifyObservers();
+  void OnComponentReady(bool is_super_referrer,
+                        const base::FilePath& installed_dir);
+  void OnGetComponentJsonData(const std::string& json_string);
+  void OnPreferenceChanged(const std::string& pref_name);
+  void OnGetMappingTableData(std::unique_ptr<std::string> json_string);
 
+  void GetComponentJsonData(const base::FilePath& installed_dir);
+  void NotifyObservers();
+  void DetermineTargetComponent();
+  bool IsSuperReferrerCode(const std::string& referral_code);
```

Comment:
why are these Referrer/Referral methods on NTPBackgroundImagesService. We should be getting these from BraveReferralsService (and we need a DependsOn for it)

Reply:
`NTPBackgroundImagesService` doesn't directly depends on `BraveReferralsService`.
It just monitors referrals code change during the runtime by observing `kReferralPromoCode` prefs.
and I think BraveReferralsService doesn't need to know about `super referral`.
Super referral has only meaning in the context of custom homepage.
To `BraveReferralsService`, super referral is just referral.
AFAIK, android doesn't use `BraveReferralsService` for handling referral code.
So, just monitoring `kReferralPromoCode` prefs will work on both (desktop and android).

Reply:
I assume that `kReferralPromoCode` is set properly from android, `NTPBackgroundImagesService` will work well. cc @deeppandya